src/collections/Menu/Menu.js

```diff
@@ -81,10 +81,11 @@ export default class Menu extends Component {
 
     return Children.map(children, (child, i) => {
       const isItem = child.type === MenuItem
+      const isLink = _.has(child, 'props.href') || _.has(child, 'props.link') || _.has(child, 'props.onClick')
 
       if (isItem) {
         const onClick = (e) => {
-          this.handleItemClick(e, i)
+          if (isLink) this.handleItemClick(e, i)
           if (child.props.onClick) child.props.onClick(e, i)
         }
```

Comment:
Take a look on basic example of [Menu](http://semantic-ui.com/collections/menu.html). After click on `item` it switches `active` class to this item. I've implemented it like it is in `Accordion`, but as you say this solution has problem.

I'm correctly understand that you suggest to use `Content`? But, as you answered earlier:
> Context is an advanced and experimental feature. The API is likely to change in future releases.